src/purify.ts

```diff
@@ -1950,6 +1953,10 @@ interface HooksMap {
   uponSanitizeAttribute: UponSanitizeAttributeHook[];
 }
 
+type ArrayElement<T> = T extends Array<infer U> ? U : never;
+
+type HookFunction = ArrayElement<HooksMap[keyof HooksMap]>;
```

Comment:
This is a utility type that extracts each element type of a union type composed of an array.

```ts
type test = ArrayElement<string[]>; // string
type test2 = ArrayElement<number[]>; // number
type test3 = ArrayElement<string[] | number[]>; // string | number
```
```ts
type HookFunction = ArrayElement<HooksMap[keyof HooksMap]>;
// NodeHook | DocumentFragmentHook | ElementHook | UponSanitizeElementHook | UponSanitizeAttributeHook
```